internal/services/openai.go

```diff
@@ -89,7 +101,17 @@ func (o openAI) askChatGPT(body io.Reader) (*ChatGPTResponse, error) {
 	return cResp, nil
 }
 
-func (o openAI) GetErrorCodesDescription(make, model string, errorCodes []string) (string, error) {
+func (o *openAI) measureAPIResponseTime(req *http.Request, start time.Time) *http.Request {
+	trace := &httptrace.ClientTrace{
+		GotFirstResponseByte: func() {
+			o.currentReqResponseTime = time.Since(start)
```

Comment:
I think attaching this to the shared OpenAI client exposes us to races while not offering much benefit. The user of `httptrace` is interesting.